File: server/app/models/minute_bar.rb
# == Schema Information
#
# Table name: minute_bars
#
#  id             :uuid             not null, primary key
#  stock_id       :uuid             not null
#  data_source    :string           not null
#  quote_time     :datetime         not null
#  high           :decimal(15, 2)
#  open           :decimal(15, 2)
#  close          :decimal(15, 2)
#  low            :decimal(15, 2)
#  adjusted_close :decimal(15, 2)
#  volume         :decimal(15, 2)
#  created_at     :datetime         not null
#  updated_at     :datetime         not null
#

class MinuteBar < ApplicationRecord
  belongs_to :stock
end

File: phoenix_api/web/models/minute_bar.ex
defmodule PhoenixApi.MinuteBar do
  use PhoenixApi.Web, :model

  schema "minute_bars" do
    belongs_to :stock, PhoenixApi.Stock
    field :data_source, :string
    field :quote_time, Ecto.DateTime
    field :high, :decimal
    field :open, :decimal
    field :close, :decimal
    field :low, :decimal
    field :adjusted_close, :decimal
    field :volume, :decimal

    timestamps()
  end

  @doc """
  Builds a changeset based on the `struct` and `params`.
  """
  def changeset(struct, params \\ %{}) do
    struct
    |> cast(params, [:stock_id, :data_source, :quote_time, :high, :open, :close, :low, :adjusted_close, :volume])
    |> validate_required([:stock_id, :data_source, :quote_time, :high, :open, :close, :low, :adjusted_close, :volume])
  end
end
